integration-tests/chain-signatures/tests/cases/mod.rs

```diff
@@ -28,191 +27,156 @@ pub mod nightly;
-        Box::pin(async move {
-            let state = wait_for::running_mpc(&ctx, Some(0)).await?;
-            wait_for::has_at_least_triples(&ctx, 2).await?;
-            wait_for::has_at_least_presignatures(&ctx, 2).await?;
-            actions::single_signature_production(&ctx, &state).await?;
-
-            tracing::info!("!!! Add participant 3");
-            assert!(ctx.add_participant(None).await.is_ok());
-            let state = wait_for::running_mpc(&ctx, None).await?;
-            wait_for::has_at_least_triples(&ctx, 2).await?;
-            wait_for::has_at_least_presignatures(&ctx, 2).await?;
-            actions::single_signature_production(&ctx, &state).await?;
-
-            tracing::info!("!!! Remove participant 0 and participant 2");
-            let account_2 = near_workspaces::types::AccountId::from_str(
-                state.participants.keys().nth(2).unwrap().clone().as_ref(),
-            )
-            .unwrap();
-            assert!(ctx.remove_participant(Some(&account_2)).await.is_ok());
-            let account_0 = near_workspaces::types::AccountId::from_str(
-                state.participants.keys().next().unwrap().clone().as_ref(),
-            )
-            .unwrap();
-            let node_cfg_0 = ctx.remove_participant(Some(&account_0)).await;
-            assert!(node_cfg_0.is_ok());
-            let node_cfg_0 = node_cfg_0.unwrap();
-            let state = wait_for::running_mpc(&ctx, None).await?;
-            wait_for::has_at_least_triples(&ctx, 2).await?;
-            wait_for::has_at_least_presignatures(&ctx, 2).await?;
-            actions::single_signature_production(&ctx, &state).await?;
-
-            tracing::info!("!!! Try remove participant 3, should fail due to threshold");
-            assert!(ctx.remove_participant(None).await.is_err());
-
-            tracing::info!("!!! Add participant 5");
-            assert!(ctx.add_participant(None).await.is_ok());
-            let state = wait_for::running_mpc(&ctx, None).await?;
-            wait_for::has_at_least_triples(&ctx, 2).await?;
-            wait_for::has_at_least_presignatures(&ctx, 2).await?;
-            actions::single_signature_production(&ctx, &state).await?;
-
-            tracing::info!("!!! Add back participant 0");
-            assert!(ctx.add_participant(Some(node_cfg_0)).await.is_ok());
-            let state = wait_for::running_mpc(&ctx, None).await?;
-            wait_for::has_at_least_triples(&ctx, 2).await?;
-            wait_for::has_at_least_presignatures(&ctx, 2).await?;
-            actions::single_signature_production(&ctx, &state).await
-        })
-    })
-    .await
+    let mut nodes = cluster::spawn().wait_for_running().await?;
+    nodes.wait().signable().await?;
+    let _ = nodes.sign().await?;
+
+    tracing::info!("!!! Add participant 3");
+    nodes.add_participant(None).await.unwrap();
+    let state = nodes.wait().running().signable().await.unwrap();
+    let _ = nodes.sign().await.unwrap();
```

Comment:
what is the reason for unwrap() instead of `?`

Reply:
unwrap will give you the backtrace as it panics meanwhile `?` will give you the anyhow error message only once its forward past `main`